monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/RepositoryZipController.java

```diff
@@ -78,28 +80,58 @@ public void downloadRepository(@PathVariable("tenant") final String tenant, Http
         FileHelper.safeClose(zipOutput);
     }
 
-    @RequestMapping(method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+    @RequestMapping(method = RequestMethod.POST, produces = MediaType.TEXT_PLAIN_VALUE)
     @ResponseBody
     @RolesAllowed(SecurityRoles.ADMIN)
-    public Map<String, String> uploadRepository(@PathVariable("tenant") final String tenant,
+    public String uploadRepository(@PathVariable("tenant") final String tenant,
             @RequestParam("file") final MultipartFile file) throws IOException {
         final TenantContext context = _tenantContextFactory.getContext(tenant);
         final RepositoryFolder rootFolder = context.getTenantRootFolder();
-
+ 
+        if (file.getSize() == 0){
+            return "Failure. The file is empty"; 
+        } 
+        
+        final String contentType = file.getContentType().trim();
+        if ((!contentType.equals("application/zip")) && (!contentType.equals("application/octet-stream"))) {
+            return "Failure. The file isn't a .zip archive";
+        }
+        
+        final File oldRepositoryZipFile = createZipfolder(rootFolder);
         logger.info("Uploading ZIP file for tenant repository: {}", tenant);
-
         try (final InputStream inputStream = file.getInputStream()) {
             ZipInputStream zipInputStream = new ZipInputStream(inputStream);
-            decompress(zipInputStream, rootFolder);
+            if (isValidRepository(zipInputStream)) {
+                decompress(zipInputStream, rootFolder);
+                return "Success.\nNew repository path: " + rootFolder.getQualifiedPath()
+                        + ".\nOld repository backed up to: " + oldRepositoryZipFile.toString();
+            } else {
+                // delete the repository if the uploading failed
+                oldRepositoryZipFile.delete();
+                return "Failure. The repository does not contain a conf.xml file";
+            }
         }
-
-        final Map<String, String> result = new HashMap<String, String>();
-        result.put("status", "Success");
-        result.put("repository_path", rootFolder.getQualifiedPath());
-
-        return result;
     }
 
+    private File createZipfolder(RepositoryFolder rootFolder) throws IOException {
+        final long timeInMillis = Calendar.getInstance().getTimeInMillis();
+        final File tempfile = File.createTempFile("repository_" + rootFolder.getName() + "_" + timeInMillis, ".zip");
+        final FileOutputStream fos = new FileOutputStream(tempfile);
+        final ZipOutputStream zipOutput = new ZipOutputStream(fos);
+        compress(rootFolder, zipOutput);
+        logger.info("The old repository has been compressed and uploaded to" + tempfile.getAbsolutePath());
+        FileHelper.safeClose(zipOutput);
+        return tempfile;
+    }
+    /** the repository is valid if it contains a conf.xml file**/
```

Comment:
Why not use javadoc style for the comment? Seems like a better fit for a comment describing what a method does.